### PR TITLE
feature/new_kiosk_nodes_33

### PR DESCRIPTION
**Pull request by sjpridotkas**
Adding kiosk class and fuctionality

### CODE REVIEW

**Comment by benhylak** [comment]
Build failed..

File: src/Boundary/AdminMapBoundary.java [hunk 1]
```diff
@@ -73,6 +76,11 @@ public MapNode newNode(DragIconType type, Point2D loc)
             }
             case Kiosk:
             {
+                Kiosk newKiosk = new Kiosk();
+                Hospital.kiosks.add(newKiosk);
+                n = newKiosk;
```
Comment (bhylak) [inline src/Boundary/AdminMapBoundary.java]: Glad you figured out why its setup like this. (has to do with how a lambda can reference things)

File: src/Boundary/AdminMapBoundary.java [hunk 2]
```diff
@@ -169,4 +177,15 @@ public void moveNode(MapNode n, Point2D movedTo)
             edge.updateCost();
         }
     }
+
+    public void setCurrentKiosk (Kiosk kiosk) {
+        if (Hospital.kiosks.contains(kiosk)) {
+            Hospital.currentKiosk = kiosk;
+        } else {
```
Comment (benhylak) [inline src/Boundary/AdminMapBoundary.java]: Is it possible to make kiosks protected and instead have a "getKiosks()" function or "containsKiosk" function

File: src/Boundary/AdminMapBoundary.java [hunk 3]
```diff
@@ -169,4 +177,15 @@ public void moveNode(MapNode n, Point2D movedTo)
...
+            Hospital.kiosks.add(kiosk);
+            Hospital.currentKiosk = kiosk;
+        }
+    }
```
Comment (benhylak) [inline src/Boundary/AdminMapBoundary.java]: Same thing here.. "setCurrentKiosk" function

File: src/Controller/Map/Admin/PopUp/KioskEditController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    protected PopOver popOver;
+
+    protected Kiosk kioskUnderEdit;
+
+    public KioskEditController(Kiosk kiosk)
+    {
+        super(kiosk);
+
+        this.kioskUnderEdit = kiosk;
+    }
```
Comment (benhylak) [inline src/Controller/Map/Admin/PopUp/KioskEditController.java:29]: Does this have  drop down for direction its facing? (N, S, E, W, etc.)

File: src/Entity/Hospital.java [hunk 5]
```diff
@@ -24,13 +25,18 @@
 
     Entity.Map.CampusFloor CampusFloor;
 
+    public static Kiosk currentKiosk = null;
```
Comment (benhylak) [inline src/Entity/Hospital.java]: should be private and not static

File: src/Entity/Hospital.java [hunk 6]
```diff
@@ -24,13 +25,18 @@
...
+    public static ObservableList<Kiosk> kiosks;
```
Comment (benhylak) [inline src/Entity/Hospital.java]: should be just "protected"

File: src/Entity/Hospital.java [hunk 7]
```diff
@@ -81,4 +87,11 @@ public void addOffices(String s, Office off) {
         return buildings;
     }
 
+    public ObservableList<Kiosk> getKiosks () { return kiosks; }
+    public Kiosk getCurrentKiosk() { return  currentKiosk; }
+
+    public void addKiosk (Kiosk newKiosk) {
+        kiosks.add(newKiosk);
```
Comment (benhylak) [inline src/Entity/Hospital.java:94]: Okay so you already have this!  Just change visibility of the variable and then switch over the boundary code to use these functions

File: src/Entity/Map/Kiosk.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    public String getName() {
+        return this.name;
+    }
+
+    public void setName(String s) {
+        this.name = s;
+    }
+
+    public Character getDirection() { return this.direction; }
+
```
Comment (benhylak) [inline src/Entity/Map/Kiosk.java:77]: Looks good

**Review by benhylak — CHANGES_REQUESTED** [review]
Just address a few simple changes, particularly the visibility of the kiosks list

**Review by benhylak — APPROVED** [review]

**Comment by iancj97** [comment]
This branch can be deleted

**Pull request merged**